golded3/geedit.cpp

```diff
@@ -2330,7 +2330,7 @@ void IEclass::SCheckerMenu()
         finaltag = menu.Run(schecker, buff);
         if (finaltag >= 0)
         {
-            std::string &str = schecker.GetSuggest()[finaltag].second;
```

Comment:
Hmm, why it was ".second" before and now it's removed?

Reply:
It was vector of pairs. And first in the pair never used.

Reply:
Why it was never used? :) Maybe there were reasons?

Reply:
I mean that member not used anywhere in the code. Both for MS and hunspell.

Reply:
It is a tuple of <byte, string>. If I understand it correctly, &str was assigned to the string part of the tuple before, now it points the the beginning of the tuple, isn't it? Shouldn't we compensate the extra byte (the first item of the tuple) in the line 2335 or somehow else? Sorry I I'm writing something stupid, c++ is not my strong part...

Reply:
Just looked at commit, when spell checker was added. Maybe author was going to use it for ordering, but all inserts into vector has that byte set to 0 and no reads for that field. There is no reason to keep that field.

Reply:
so, in order to keep functionality as it was before and don't break it with some unexpected 0 at the beginning of some buffer, shouldn't we replace the tuple with just string?

Reply:
It's not a pure c. :) That 0 byte was never used.
In std::pair<byte, string> both fields are totally separate members.

Reply:
Sorry, I haven't found that you actually replaced the type implementation behind. I looked again, and now I see it. No questions anymore :)